// @adyen/api-library/lib/src/typings/platformsAccount/createAccountResponse.js
"use strict";
/**
 * Adyen for Platforms: Account API
 * The Account API provides endpoints for managing account-related entities on your platform. These related entities include account holders, accounts, bank accounts, shareholders, and KYC-related documents. The management operations include actions such as creation, retrieval, updating, and deletion of them.  For more information, refer to our [documentation](https://docs.adyen.com/platforms). ## Authentication To connect to the Account API, you must use basic authentication credentials of your web service user. If you don\'t have one, contact the [Adyen Support Team](https://support.adyen.com/hc/en-us/requests/new). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@MarketPlace.YourMarketPlace\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning The Account API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://cal-test.adyen.com/cal/services/Account/v6/createAccountHolder ```
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAccountResponse = void 0;
class CreateAccountResponse {
    static getAttributeTypeMap() {
        return CreateAccountResponse.attributeTypeMap;
    }
}
exports.CreateAccountResponse = CreateAccountResponse;
CreateAccountResponse.discriminator = undefined;
CreateAccountResponse.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "PayoutScheduleResponse"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "CreateAccountResponse.PayoutSpeedEnum"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CreateAccountResponse.StatusEnum"
    }
];
(function (CreateAccountResponse) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum[PayoutSpeedEnum["Instant"] = 'INSTANT'] = "Instant";
        PayoutSpeedEnum[PayoutSpeedEnum["SameDay"] = 'SAME_DAY'] = "SameDay";
        PayoutSpeedEnum[PayoutSpeedEnum["Standard"] = 'STANDARD'] = "Standard";
    })(PayoutSpeedEnum = CreateAccountResponse.PayoutSpeedEnum || (CreateAccountResponse.PayoutSpeedEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'Active'] = "Active";
        StatusEnum[StatusEnum["Closed"] = 'Closed'] = "Closed";
        StatusEnum[StatusEnum["Inactive"] = 'Inactive'] = "Inactive";
        StatusEnum[StatusEnum["Suspended"] = 'Suspended'] = "Suspended";
    })(StatusEnum = CreateAccountResponse.StatusEnum || (CreateAccountResponse.StatusEnum = {}));
})(CreateAccountResponse = exports.CreateAccountResponse || (exports.CreateAccountResponse = {}));
