// @adyen/api-library/lib/src/typings/platformsAccount/documentDetail.js
"use strict";
/**
 * Adyen for Platforms: Account API
 * The Account API provides endpoints for managing account-related entities on your platform. These related entities include account holders, accounts, bank accounts, shareholders, and KYC-related documents. The management operations include actions such as creation, retrieval, updating, and deletion of them.  For more information, refer to our [documentation](https://docs.adyen.com/platforms). ## Authentication To connect to the Account API, you must use basic authentication credentials of your web service user. If you don\'t have one, contact the [Adyen Support Team](https://support.adyen.com/hc/en-us/requests/new). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@MarketPlace.YourMarketPlace\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning The Account API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://cal-test.adyen.com/cal/services/Account/v6/createAccountHolder ```
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentDetail = void 0;
class DocumentDetail {
    static getAttributeTypeMap() {
        return DocumentDetail.attributeTypeMap;
    }
}
exports.DocumentDetail = DocumentDetail;
DocumentDetail.discriminator = undefined;
DocumentDetail.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "documentType",
        "baseName": "documentType",
        "type": "DocumentDetail.DocumentTypeEnum"
    },
    {
        "name": "filename",
        "baseName": "filename",
        "type": "string"
    },
    {
        "name": "shareholderCode",
        "baseName": "shareholderCode",
        "type": "string"
    },
    {
        "name": "signatoryCode",
        "baseName": "signatoryCode",
        "type": "string"
    }
];
(function (DocumentDetail) {
    let DocumentTypeEnum;
    (function (DocumentTypeEnum) {
        DocumentTypeEnum[DocumentTypeEnum["BankStatement"] = 'BANK_STATEMENT'] = "BankStatement";
        DocumentTypeEnum[DocumentTypeEnum["Bsn"] = 'BSN'] = "Bsn";
        DocumentTypeEnum[DocumentTypeEnum["CompanyRegistrationScreening"] = 'COMPANY_REGISTRATION_SCREENING'] = "CompanyRegistrationScreening";
        DocumentTypeEnum[DocumentTypeEnum["ConstitutionalDocument"] = 'CONSTITUTIONAL_DOCUMENT'] = "ConstitutionalDocument";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicence"] = 'DRIVING_LICENCE'] = "DrivingLicence";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicenceBack"] = 'DRIVING_LICENCE_BACK'] = "DrivingLicenceBack";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicenceFront"] = 'DRIVING_LICENCE_FRONT'] = "DrivingLicenceFront";
        DocumentTypeEnum[DocumentTypeEnum["IdCard"] = 'ID_CARD'] = "IdCard";
        DocumentTypeEnum[DocumentTypeEnum["IdCardBack"] = 'ID_CARD_BACK'] = "IdCardBack";
        DocumentTypeEnum[DocumentTypeEnum["IdCardFront"] = 'ID_CARD_FRONT'] = "IdCardFront";
        DocumentTypeEnum[DocumentTypeEnum["Passport"] = 'PASSPORT'] = "Passport";
        DocumentTypeEnum[DocumentTypeEnum["Ssn"] = 'SSN'] = "Ssn";
        DocumentTypeEnum[DocumentTypeEnum["SupportingDocuments"] = 'SUPPORTING_DOCUMENTS'] = "SupportingDocuments";
    })(DocumentTypeEnum = DocumentDetail.DocumentTypeEnum || (DocumentDetail.DocumentTypeEnum = {}));
})(DocumentDetail = exports.DocumentDetail || (exports.DocumentDetail = {}));
