// @adyen/api-library/lib/src/typings/platformsAccount/kYCCheckStatusData.js
"use strict";
/**
 * Adyen for Platforms: Account API
 * The Account API provides endpoints for managing account-related entities on your platform. These related entities include account holders, accounts, bank accounts, shareholders, and KYC-related documents. The management operations include actions such as creation, retrieval, updating, and deletion of them.  For more information, refer to our [documentation](https://docs.adyen.com/platforms). ## Authentication To connect to the Account API, you must use basic authentication credentials of your web service user. If you don\'t have one, contact the [Adyen Support Team](https://support.adyen.com/hc/en-us/requests/new). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@MarketPlace.YourMarketPlace\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning The Account API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://cal-test.adyen.com/cal/services/Account/v6/createAccountHolder ```
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KYCCheckStatusData = void 0;
class KYCCheckStatusData {
    static getAttributeTypeMap() {
        return KYCCheckStatusData.attributeTypeMap;
    }
}
exports.KYCCheckStatusData = KYCCheckStatusData;
KYCCheckStatusData.discriminator = undefined;
KYCCheckStatusData.attributeTypeMap = [
    {
        "name": "requiredFields",
        "baseName": "requiredFields",
        "type": "Array<string>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "KYCCheckStatusData.StatusEnum"
    },
    {
        "name": "summary",
        "baseName": "summary",
        "type": "KYCCheckSummary"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "KYCCheckStatusData.TypeEnum"
    }
];
(function (KYCCheckStatusData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AwaitingData"] = 'AWAITING_DATA'] = "AwaitingData";
        StatusEnum[StatusEnum["DataProvided"] = 'DATA_PROVIDED'] = "DataProvided";
        StatusEnum[StatusEnum["Failed"] = 'FAILED'] = "Failed";
        StatusEnum[StatusEnum["InvalidData"] = 'INVALID_DATA'] = "InvalidData";
        StatusEnum[StatusEnum["Passed"] = 'PASSED'] = "Passed";
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["PendingReview"] = 'PENDING_REVIEW'] = "PendingReview";
        StatusEnum[StatusEnum["RetryLimitReached"] = 'RETRY_LIMIT_REACHED'] = "RetryLimitReached";
        StatusEnum[StatusEnum["Unchecked"] = 'UNCHECKED'] = "Unchecked";
    })(StatusEnum = KYCCheckStatusData.StatusEnum || (KYCCheckStatusData.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BankAccountVerification"] = 'BANK_ACCOUNT_VERIFICATION'] = "BankAccountVerification";
        TypeEnum[TypeEnum["CardVerification"] = 'CARD_VERIFICATION'] = "CardVerification";
        TypeEnum[TypeEnum["CompanyVerification"] = 'COMPANY_VERIFICATION'] = "CompanyVerification";
        TypeEnum[TypeEnum["IdentityVerification"] = 'IDENTITY_VERIFICATION'] = "IdentityVerification";
        TypeEnum[TypeEnum["LegalArrangementVerification"] = 'LEGAL_ARRANGEMENT_VERIFICATION'] = "LegalArrangementVerification";
        TypeEnum[TypeEnum["NonprofitVerification"] = 'NONPROFIT_VERIFICATION'] = "NonprofitVerification";
        TypeEnum[TypeEnum["PassportVerification"] = 'PASSPORT_VERIFICATION'] = "PassportVerification";
        TypeEnum[TypeEnum["PayoutMethodVerification"] = 'PAYOUT_METHOD_VERIFICATION'] = "PayoutMethodVerification";
        TypeEnum[TypeEnum["PciVerification"] = 'PCI_VERIFICATION'] = "PciVerification";
    })(TypeEnum = KYCCheckStatusData.TypeEnum || (KYCCheckStatusData.TypeEnum = {}));
})(KYCCheckStatusData = exports.KYCCheckStatusData || (exports.KYCCheckStatusData = {}));
