// @adyen/api-library/lib/src/typings/platformsAccount/legalArrangementDetail.js
"use strict";
/**
 * Adyen for Platforms: Account API
 * The Account API provides endpoints for managing account-related entities on your platform. These related entities include account holders, accounts, bank accounts, shareholders, and KYC-related documents. The management operations include actions such as creation, retrieval, updating, and deletion of them.  For more information, refer to our [documentation](https://docs.adyen.com/platforms). ## Authentication To connect to the Account API, you must use basic authentication credentials of your web service user. If you don\'t have one, contact the [Adyen Support Team](https://support.adyen.com/hc/en-us/requests/new). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@MarketPlace.YourMarketPlace\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning The Account API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://cal-test.adyen.com/cal/services/Account/v6/createAccountHolder ```
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementDetail = void 0;
class LegalArrangementDetail {
    static getAttributeTypeMap() {
        return LegalArrangementDetail.attributeTypeMap;
    }
}
exports.LegalArrangementDetail = LegalArrangementDetail;
LegalArrangementDetail.discriminator = undefined;
LegalArrangementDetail.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "legalArrangementCode",
        "baseName": "legalArrangementCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntities",
        "baseName": "legalArrangementEntities",
        "type": "Array<LegalArrangementEntityDetail>"
    },
    {
        "name": "legalArrangementReference",
        "baseName": "legalArrangementReference",
        "type": "string"
    },
    {
        "name": "legalForm",
        "baseName": "legalForm",
        "type": "LegalArrangementDetail.LegalFormEnum"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "taxNumber",
        "baseName": "taxNumber",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalArrangementDetail.TypeEnum"
    }
];
(function (LegalArrangementDetail) {
    let LegalFormEnum;
    (function (LegalFormEnum) {
        LegalFormEnum[LegalFormEnum["CashManagementTrust"] = 'CashManagementTrust'] = "CashManagementTrust";
        LegalFormEnum[LegalFormEnum["CorporateUnitTrust"] = 'CorporateUnitTrust'] = "CorporateUnitTrust";
        LegalFormEnum[LegalFormEnum["DeceasedEstate"] = 'DeceasedEstate'] = "DeceasedEstate";
        LegalFormEnum[LegalFormEnum["DiscretionaryInvestmentTrust"] = 'DiscretionaryInvestmentTrust'] = "DiscretionaryInvestmentTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryServicesManagementTrust"] = 'DiscretionaryServicesManagementTrust'] = "DiscretionaryServicesManagementTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryTradingTrust"] = 'DiscretionaryTradingTrust'] = "DiscretionaryTradingTrust";
        LegalFormEnum[LegalFormEnum["FirstHomeSaverAccountsTrust"] = 'FirstHomeSaverAccountsTrust'] = "FirstHomeSaverAccountsTrust";
        LegalFormEnum[LegalFormEnum["FixedTrust"] = 'FixedTrust'] = "FixedTrust";
        LegalFormEnum[LegalFormEnum["FixedUnitTrust"] = 'FixedUnitTrust'] = "FixedUnitTrust";
        LegalFormEnum[LegalFormEnum["HybridTrust"] = 'HybridTrust'] = "HybridTrust";
        LegalFormEnum[LegalFormEnum["ListedPublicUnitTrust"] = 'ListedPublicUnitTrust'] = "ListedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["OtherTrust"] = 'OtherTrust'] = "OtherTrust";
        LegalFormEnum[LegalFormEnum["PooledSuperannuationTrust"] = 'PooledSuperannuationTrust'] = "PooledSuperannuationTrust";
        LegalFormEnum[LegalFormEnum["PublicTradingTrust"] = 'PublicTradingTrust'] = "PublicTradingTrust";
        LegalFormEnum[LegalFormEnum["UnlistedPublicUnitTrust"] = 'UnlistedPublicUnitTrust'] = "UnlistedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["LimitedPartnership"] = 'LimitedPartnership'] = "LimitedPartnership";
        LegalFormEnum[LegalFormEnum["FamilyPartnership"] = 'FamilyPartnership'] = "FamilyPartnership";
        LegalFormEnum[LegalFormEnum["OtherPartnership"] = 'OtherPartnership'] = "OtherPartnership";
    })(LegalFormEnum = LegalArrangementDetail.LegalFormEnum || (LegalArrangementDetail.LegalFormEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Partnership"] = 'Partnership'] = "Partnership";
        TypeEnum[TypeEnum["Trust"] = 'Trust'] = "Trust";
    })(TypeEnum = LegalArrangementDetail.TypeEnum || (LegalArrangementDetail.TypeEnum = {}));
})(LegalArrangementDetail = exports.LegalArrangementDetail || (exports.LegalArrangementDetail = {}));
