// @adyen/api-library/lib/src/typings/platformsAccount/payoutScheduleResponse.js
"use strict";
/**
 * Adyen for Platforms: Account API
 * The Account API provides endpoints for managing account-related entities on your platform. These related entities include account holders, accounts, bank accounts, shareholders, and KYC-related documents. The management operations include actions such as creation, retrieval, updating, and deletion of them.  For more information, refer to our [documentation](https://docs.adyen.com/platforms). ## Authentication To connect to the Account API, you must use basic authentication credentials of your web service user. If you don\'t have one, contact the [Adyen Support Team](https://support.adyen.com/hc/en-us/requests/new). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@MarketPlace.YourMarketPlace\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning The Account API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://cal-test.adyen.com/cal/services/Account/v6/createAccountHolder ```
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutScheduleResponse = void 0;
class PayoutScheduleResponse {
    static getAttributeTypeMap() {
        return PayoutScheduleResponse.attributeTypeMap;
    }
}
exports.PayoutScheduleResponse = PayoutScheduleResponse;
PayoutScheduleResponse.discriminator = undefined;
PayoutScheduleResponse.attributeTypeMap = [
    {
        "name": "nextScheduledPayout",
        "baseName": "nextScheduledPayout",
        "type": "Date"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "PayoutScheduleResponse.ScheduleEnum"
    }
];
(function (PayoutScheduleResponse) {
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT'] = "BiweeklyOn1StAnd15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtNoon"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_NOON'] = "BiweeklyOn1StAnd15ThAtNoon";
        ScheduleEnum[ScheduleEnum["Daily"] = 'DAILY'] = "Daily";
        ScheduleEnum[ScheduleEnum["Daily6Pm"] = 'DAILY_6PM'] = "Daily6Pm";
        ScheduleEnum[ScheduleEnum["DailyAu"] = 'DAILY_AU'] = "DailyAu";
        ScheduleEnum[ScheduleEnum["DailyEu"] = 'DAILY_EU'] = "DailyEu";
        ScheduleEnum[ScheduleEnum["DailySg"] = 'DAILY_SG'] = "DailySg";
        ScheduleEnum[ScheduleEnum["DailyUs"] = 'DAILY_US'] = "DailyUs";
        ScheduleEnum[ScheduleEnum["Default"] = 'DEFAULT'] = "Default";
        ScheduleEnum[ScheduleEnum["Every6HoursFromMidnight"] = 'EVERY_6_HOURS_FROM_MIDNIGHT'] = "Every6HoursFromMidnight";
        ScheduleEnum[ScheduleEnum["Hold"] = 'HOLD'] = "Hold";
        ScheduleEnum[ScheduleEnum["Monthly"] = 'MONTHLY'] = "Monthly";
        ScheduleEnum[ScheduleEnum["MonthlyOn15ThAtMidnight"] = 'MONTHLY_ON_15TH_AT_MIDNIGHT'] = "MonthlyOn15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["Weekly"] = 'WEEKLY'] = "Weekly";
        ScheduleEnum[ScheduleEnum["WeeklyOnTueFriMidnight"] = 'WEEKLY_ON_TUE_FRI_MIDNIGHT'] = "WeeklyOnTueFriMidnight";
        ScheduleEnum[ScheduleEnum["Yearly"] = 'YEARLY'] = "Yearly";
    })(ScheduleEnum = PayoutScheduleResponse.ScheduleEnum || (PayoutScheduleResponse.ScheduleEnum = {}));
})(PayoutScheduleResponse = exports.PayoutScheduleResponse || (exports.PayoutScheduleResponse = {}));
