// @adyen/api-library/lib/src/typings/recurring/recurring.js
"use strict";
/**
 * Adyen Recurring API
 * The Recurring APIs allow you to manage and remove your tokens or saved payment details. Tokens should be created with validation during a payment request.  For more information, refer to our [Tokenization documentation](https://docs.adyen.com/online-payments/tokenization). ## Authentication To connect to the Recurring API, you must use your basic authentication credentials. For this, create your web service user, as described in [How to get the WS user password](https://docs.adyen.com/development-resources/api-credentials). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@Company.YourCompany\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning Recurring API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://pal-test.adyen.com/pal/servlet/Recurring/v49/disable ```
 *
 * The version of the OpenAPI document: 49
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recurring = void 0;
class Recurring {
    static getAttributeTypeMap() {
        return Recurring.attributeTypeMap;
    }
}
exports.Recurring = Recurring;
Recurring.discriminator = undefined;
Recurring.attributeTypeMap = [
    {
        "name": "contract",
        "baseName": "contract",
        "type": "Recurring.ContractEnum"
    },
    {
        "name": "recurringDetailName",
        "baseName": "recurringDetailName",
        "type": "string"
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "Date"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string"
    },
    {
        "name": "tokenService",
        "baseName": "tokenService",
        "type": "Recurring.TokenServiceEnum"
    }
];
(function (Recurring) {
    let ContractEnum;
    (function (ContractEnum) {
        ContractEnum[ContractEnum["Oneclick"] = 'ONECLICK'] = "Oneclick";
        ContractEnum[ContractEnum["Recurring"] = 'RECURRING'] = "Recurring";
        ContractEnum[ContractEnum["Payout"] = 'PAYOUT'] = "Payout";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    let TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum[TokenServiceEnum["Visatokenservice"] = 'VISATOKENSERVICE'] = "Visatokenservice";
        TokenServiceEnum[TokenServiceEnum["Mctokenservice"] = 'MCTOKENSERVICE'] = "Mctokenservice";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring = exports.Recurring || (exports.Recurring = {}));
