// @adyen/api-library/lib/src/typings/checkout/split.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Split = void 0;
class Split {
    static getAttributeTypeMap() {
        return Split.attributeTypeMap;
    }
}
exports.Split = Split;
Split.discriminator = undefined;
Split.attributeTypeMap = [
    {
        "name": "account",
        "baseName": "account",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "SplitAmount"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Split.TypeEnum"
    }
];
(function (Split) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BalanceAccount"] = 'BalanceAccount'] = "BalanceAccount";
        TypeEnum[TypeEnum["Commission"] = 'Commission'] = "Commission";
        TypeEnum[TypeEnum["Default"] = 'Default'] = "Default";
        TypeEnum[TypeEnum["MarketPlace"] = 'MarketPlace'] = "MarketPlace";
        TypeEnum[TypeEnum["PaymentFee"] = 'PaymentFee'] = "PaymentFee";
        TypeEnum[TypeEnum["Vat"] = 'VAT'] = "Vat";
        TypeEnum[TypeEnum["Verification"] = 'Verification'] = "Verification";
    })(TypeEnum = Split.TypeEnum || (Split.TypeEnum = {}));
})(Split = exports.Split || (exports.Split = {}));
